modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/api/IgniteSqlImpl.java

```diff
@@ -29,20 +39,38 @@
 /**
  * Embedded implementation of the Ignite SQL query facade.
  */
-public class IgniteSqlImpl implements IgniteSql {
+public class IgniteSqlImpl implements IgniteSql, IgniteComponent {
+    private static final IgniteLogger LOG = Loggers.forClass(IgniteSqlImpl.class);
+
+    /** Session expiration check period in milliseconds. */
+    private static final long SESSION_EXPIRE_CHECK_PERIOD = TimeUnit.SECONDS.toMillis(1);
+
     private final QueryProcessor qryProc;
 
     private final IgniteTransactions transactions;
 
+    /** Session expiration worker. */
+    private final ScheduledExecutorService executor;
+
+    private final ConcurrentMap<SessionId, SessionImpl> sessions = new ConcurrentHashMap<>();
+
     /**
      * Constructor.
      *
      * @param qryProc Query processor.
      * @param transactions Transactions facade.
      */
-    public IgniteSqlImpl(QueryProcessor qryProc, IgniteTransactions transactions) {
+    public IgniteSqlImpl(
+            String nodeName,
```

Comment:
I don't like the idea to pass node name to components just to create thread pool.
Also, with this approach, we can't use spy on thread factory in test to verify the component stops correctly.

WDYT, if we will have NodeThreadFactory and pass the factory to components, instead of nodeName?
Feel free keeping changes as is, we always can fix this another ticket.

Reply:
Maybe it make sense to have shared scheduler thead and reuse it here.

Reply:
I was thinking about using a common scheduler, but we don't have common pools yet